File: go/0404E-左叶子之和.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-04 14:21:19
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumOfLeftLeaves(root *TreeNode) int {
	ans := 0
	var dfs func(node *TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}
		if node.Left != nil && node.Left.Left == nil && node.Left.Right == nil {
			ans += node.Left.Val
		}
		dfs(node.Left)
		dfs(node.Right)
	}
	dfs(root)
	return ans
}

File: cpp/0404E-左叶子之和.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-21 10:43:17
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int ans = 0;

    void dfs(TreeNode* root) {
        if (root == nullptr) return;
        if (root->left != nullptr && root->left->left == nullptr &&
            root->left->right == nullptr)
            ans += root->left->val;
        dfs(root->left);
        dfs(root->right);
    }

    int sumOfLeftLeaves(TreeNode* root) {
        dfs(root);
        return ans;
    }
};